doc/user/content/ops/optimization.md

```diff
@@ -55,37 +59,146 @@ In general, your index key should exactly match the columns that are constrained
 
 ### `JOIN`
-Speed up a `JOIN` query by indexing the join keys:
 
-Clause                                      | Index                                                                       |
---------------------------------------------|-----------------------------------------------------------------------------|
-`FROM view V JOIN table T ON (V.id = T.id)` | `CREATE INDEX ON view (id);` <br /> `CREATE INDEX ON table (id);`           |
+In general, you can improve the performance of your joins by creating indexes on the columns being joined. This comes at the cost of additional memory usage. Fortunately, Materialize can reuse these indexes for different queries, which means **for multiple queries, indexes are a fixed upfront cost with memory savings for each new query.**
+
+Let's create a few tables to work through examples.
+
+```sql
+CREATE TABLE teachers (id INT, name TEXT);
+CREATE TABLE sections (id INT, teacher_id INT, course_id INT, schedule TEXT);
+CREATE TABLE courses (id INT, name TEXT);
+```
+
+#### Joining Two Collections
+
+Here is an example where we join a collection `teachers` to a collection `sections` to see the name of the teacher, schedule, and course ID for a specific section of a course.
+
+```sql
+SELECT
+    t.name,
+    s.schedule,
+    s.course_id
+FROM teachers t
+INNER JOIN sections s ON (t.id = s.teacher_id);
+```
+
+We can optimize this query by creating an index for each column being joined.
+
+```sql
+CREATE INDEX pk_teachers ON teachers (id);
+CREATE INDEX sections_fk_teachers ON sections (teacher_id);
+```
+
+#### Joins with Filters
 
-For joins between more than two inputs, you should strive for a *delta join* (a special join implementation of Materialize for making multi-way streaming joins memory-efficient), which is typically possible when you index all the join keys:
+If your query filters one or more of the join inputs by a literal equality (e.g., `WHERE t.name = 'Escalante'`), place one of those input collections first in the `FROM` clause. In particular, this can speed up [ad hoc `SELECT` queries](/sql/select/#ad-hoc-queries) by accessing collections using index lookups rather than full scans.
+
+#### Optimize Multi-Way Joins with Delta Joins
+
+Materialize has access to a join execution strategy we call `DeltaQuery`, a.k.a. **delta joins**, that aggressively re-uses indexes and maintains **_zero_** intermediate results. Materialize considers this plan only if all the necessary indexes already exist, in which case the **_additional_** memory cost of the join is **zero**. This is typically possible when you **index all the join keys**.
+
+From the previous example, suppose we want to add the name of the course rather than just the course ID. We will create a view `course_schedule` to reflect this:
```

Comment:
```suggestion
Materialize has access to a join execution strategy we call `DeltaQuery`, a.k.a. **delta joins**, that aggressively re-uses indexes and maintains **_zero_** intermediate results. Materialize considers this plan only if all the necessary indexes already exist, in which case the **_additional_** memory cost of the join is **zero**. This is typically possible when you **index all the join keys**.

From the previous example, add the name of the course rather than just the course ID reflected in a new view `course_schedule`:
```